deploy/setup.py

```diff
@@ -20,7 +20,7 @@
     package_data={"determined-deploy": [str(version_file)]},
     install_requires=[
         "requests>=2.20.0",
-        "docker>=3.7.3",
+        "docker[ssh]>=3.7.3",  # ssh extra prevents paramiko dependency error
```

Comment:
this solution doesn't really make sense to me

```    # Only required when connecting using the ssh:// protocol
    'ssh': ['paramiko>=2.4.2'],```

we aren't using this functionality? Where is this  dependency coming from?

Reply:
Not totally sure, but I am thinking it's a docker-py bug.  basically paramiko being [here](https://github.com/docker/docker-py/blob/master/requirements.txt#L10), but only specified in setup.py via that extra.

This has bitten some other projects, e.g. confluent added the transitive dependency explicitly [here](https://github.com/confluentinc/confluent-docker-utils/pull/13)

Reply:
Seems to be the compose dependency actually

Reply:
https://github.com/docker/compose/blob/master/setup.py#L40

Reply:
oh, given that, why aren't we pulling in paramiko then?  docker-compose dependency tree:
```
docker-compose==1.25.5
  - cached-property [required: >=1.2.0,<2, installed: 1.5.1]
  - docker [required: >=3.7.0,<5, installed: 4.2.0]
    - requests [required: >=2.14.2,!=2.18.0, installed: 2.23.0]
      - certifi [required: >=2017.4.17, installed: 2020.4.5.1]
      - chardet [required: >=3.0.2,<4, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.9]
    - six [required: >=1.4.0, installed: 1.14.0]
    - websocket-client [required: >=0.32.0, installed: 0.57.0]
      - six [required: Any, installed: 1.14.0]
```